flink-runtime/src/main/java/org/apache/flink/runtime/util/JvmUtils.java

```diff
@@ -56,6 +64,39 @@ public static Optional<ThreadInfoSample> createThreadInfoSample(
         return ThreadInfoSample.from(threadMxBean.getThreadInfo(threadId, maxStackTraceDepth));
     }
 
+    /**
+     * Creates a {@link ThreadInfoSample} for a specific thread. Contains thread traces if
+     * maxStackTraceDepth > 0.
+     *
+     * @param threadIds The IDs of the threads to create the thread dump for.
+     * @param maxStackTraceDepth The maximum number of entries in the stack trace to be collected.
+     * @return The thread information for the requested thread IDs.
+     */
+    public static Collection<ThreadInfoSample> createThreadInfoSample(
+            Collection<Long> threadIds, int maxStackTraceDepth) {
+        ThreadMXBean threadMxBean = ManagementFactory.getThreadMXBean();
+        long[] threadIdsArray = threadIds.stream().mapToLong(l -> l).toArray();
+
+        ThreadInfo[] threadInfo = threadMxBean.getThreadInfo(threadIdsArray, maxStackTraceDepth);
+
+        List<ThreadInfo> threadInfoNoNulls =
+                IntStream.range(0, threadIdsArray.length)
+                        .filter(
+                                i -> {
+                                    if (threadInfo[i] == null) {
+                                        LOG.debug(
```

Comment:
Nit: Not use if the sampling path is considered performance-critical but you can add 
```java
if (log.isDebugEnabled()) {
...
            }
```